sklearn/neighbors/_base.py

```diff
@@ -619,6 +619,29 @@ def _fit(self, X, y=None):
                 else:
                     self._fit_method = "brute"
 
+        if (
+            # TODO(1.3): remove "wminkowski"
+            self.effective_metric_ in ("wminkowski", "minkowski")
+            and self.effective_metric_params_["p"] < 1
+        ):
+            # For 0 < p < 1 Minkowski distances aren't valid distance
+            # metric as they do not satisfy triangular inequality:
+            # they are semimetrics.
+            # algo="kd_tree" and algo="ball_tree" can't be used because
+            # KDTree and BallTree require a proper distance metric to work properly.
+            # However, the brute-force algorithm supports semi-metrics.
+            if self._fit_method == "brute":
+                warnings.warn(
+                    "Mind that for 0 < p < 1, Minkowski metrics are not distance"
+                    " metrics. Continuing the execution with `algo='brute'`."
+                )
+            else:  # self._fit_method in ("kd_tree", "ball_tree")
+                raise ValueError(
+                    f'algo="{self._fit_method}" does not support 0 < p < 1 for '
```

Comment:
Actually, the parameter is called `algorithm` and not `algo`. The error message is currently confusing.

```suggestion
                    f'algorithm="{self._fit_method}" does not support 0 < p < 1 for '
```

Reply:
:raised_back_of_hand: I am the culprit for using "algo" instead of "algorithm".